<?xml version="1.0"?>
<launch>

	<!-- arguments -->
	<arg name="domain_path"	default="$(find hiprl_replicate)/pddl/hiprl_mini.pddl" />
	<arg name="problem_path"	default="$(find hiprl_replicate)/pddl/hiprl_problem0.pddl" />
	<arg name="main_rate" default="10"/>
	
	<!-- knowledge base -->
	<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
		<param name="domain_path" value="$(arg domain_path)" />
		<param name="problem_path" value="$(arg problem_path)" />
		<!-- conditional planning flags -->
		<param name="use_unknowns" value="false" />
	</node>

	<!-- problem generation -->
	<include file="$(find rosplan_planning_system)/launch/includes/problem_interface.launch">
		<arg name="knowledge_base"   value="rosplan_knowledge_base" />
		<arg name="domain_path"      value="$(arg domain_path)" />
		<arg name="problem_path"     value="$(find hiprl_replicate)/pddl/problem.pddl" />
		<arg name="problem_topic"    value="problem_instance" />
	</include>
	
	
	<node name="rosplan_knowledge_update_node" pkg="hiprl_replicate" type="knowledge_update_node" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base"/>
	</node>
	
</launch>

	<!-- sensing interface -->
<!-- <node name="rosplan_sensing_interface" pkg="rosplan_sensing_interface" type="sensing_interface.py" respawn="false" output="screen">  -->
<!--		<rosparam command="load" file="$(find rosplan_demos)/common/hiprl_sensing.yaml"/>-->
<!--		<param name="main_rate" value="$(arg main_rate)"/>-->
<!--	</node> -->
